### PR TITLE
Support installing homebrew packages as dependencies

### PR DESCRIPTION
**Pull request by johnewart**
Also provide a checksum for the build manifest's build dependencies and a -deps-only flag for build

### CODE REVIEW

File: buildpacks/homebrew.go [hunk 1]
```diff
@@ -44,15 +61,51 @@ func (bt HomebrewBuildTool) InstallDir() string {
...
 	case "darwin":
-		return bt.InstallDarwin()
+		err = bt.InstallDarwin()
 	case "linux":
-		return bt.InstallLinux()
+		err = bt.InstallLinux()
 	default:
-		fmt.Printf("Unsupported platform: %s\n", gi.GoOS)
-		return nil
+		err = fmt.Errorf("Unsupported platform: %s\n", gi.GoOS)
```
Comment (nictuku) [inline buildpacks/homebrew.go]: no need for newline in Errorf, same below

Comment (nictuku) [reply buildpacks/homebrew.go]: ..and better not have it

File: types/build_manifest.go [hunk 2]
```diff
@@ -1,9 +1,23 @@
 package types
 
 import (
+	"bytes"
+	"crypto/sha256"
 	"fmt"
 )
 
+const PKG_CHECKSUM_LENGTH = 12
```
Comment (nictuku) [inline types/build_manifest.go]: PkgChecksumLength, constants are usually CamelCase.

File: buildpacks/homebrew.go (hunk 1 above)
Comment (johnewart) [reply buildpacks/homebrew.go]: Fixed -- conversion from Printf artifact

File: types/build_manifest.go (hunk 2 above)
Comment (johnewart) [reply types/build_manifest.go]: Done

**Pull request merged**